File: frontend/src/main/frontend/src/app/shared/model/domain/Medication.ts
import {Ingredient} from './Ingredient';
import {MedicationType} from './MedicationType.enum';
import {Type} from 'class-transformer';

export class Medication
{
  id: number;
  name: string;
  type: MedicationType;

  @Type(() => Ingredient)
  ingredients: Ingredient[];

  constructor();
  constructor(id: number, name: string, type: MedicationType, ingredients: Ingredient[]);

  constructor(id?: number, name?: string, type?: MedicationType, ingredients?: Ingredient[])
  {
    this.id = id;
    this.name = name;
    this.type = type;
    this.ingredients = ingredients;
  }
}

File: backend/src/main/java/rs/ac/uns/ftn/sbz/backend/model/Medication.java
package rs.ac.uns.ftn.sbz.backend.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rs.ac.uns.ftn.sbz.backend.model.enumeration.MedicationType;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "medication")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Medication
{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "medication_id")
    private Long id;

    @Column(name = "medication_name", unique = true)
    private String name;

    @Column(name = "medication_type")
    @Enumerated(EnumType.STRING)
    private MedicationType type;

    @ManyToMany
    @JoinTable(name = "medication_ingredient",
               joinColumns = { @JoinColumn(name = "medication_id") },
               inverseJoinColumns = { @JoinColumn(name = "ingredient_id") })
    private Set<Ingredient> ingredients;


    public Boolean hasIngredient(Ingredient ingredient)
    {
        return this.ingredients.contains(ingredient);
    }
}
